libr/anal/p/anal_arm_cs.c

```diff
@@ -1598,7 +1598,7 @@ r4,r5,r6,3,sp,[*],12,sp,+=
 		break;
 	case ARM_INS_BL:
 	case ARM_INS_BLX:
-		r_strbuf_appendf (&op->esil, "4,pc,+,lr,=,%s,pc,=", ARG(0));
+		r_strbuf_appendf (&op->esil, "4,pc,+,lr,=,4,lr,-=,%s,pc,=", ARG(0));
```

Comment:
isn't this redundant? why not just
```
r_strbuf_appendf (&op->esil, "pc,lr,=,%s,pc,=", ARG(0));
```
it would be consistent with arm64 implementation.

and yeah all of this assume that `pc` is already pointing to the next instruction.

Reply:
See the message here https://github.com/radare/radare2/issues/10633

Reply:
```
Not sure why this is occuring because when I ae the command it produces the expected result but when I single step it does not.
Example:
PC = 0x800199a
BL 0x8001CA5
=> This causes LR to be 0x80019A2 when it should be 0x800199E
```

Reply:
ok so probably there's a wrong assumption on what the value of `pc` should be when stepping on the ESIL code